src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -377,6 +384,99 @@ public void WriteValue(string value)
             this.FlushIfBufferThresholdReached();
         }
 
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            int chunkSize = bufferWriter.FreeCapacity;
+

```

Comment:
Before writing the string to the output buffer, we should write the double quote. Maybe we should do that before flushing to prepare for the write. We should also commit the contents of the json write to the buffer writer before we start writing to the buffer writer manually:

```c#
this.CommitUtf8JsonWriterContentsToBuffer();
this.bufferWriter.Write(DoubleQuote.Slice(0, 1).Span;
this.Flush();

int chunkSize = LargeStringChunkSize;
// ... write the chunks in the loop

this.bufferWriter.Write(DoubleQuote);

// signal to the writer that we may need to write separator before the next property or value. See how we do it in `WriteRawValueCore`

```

The `DoubleQuote` definition looks like:

```c#
private readonly static ReadOnlyMemory<byte> doubleQuote = new byte[] { (byte)'"' };
```